### PR TITLE
Issue: 6975 Upgrading jackson-databind, micrometer and woodstox libraries

### PR DESCRIPTION
**Pull request by anju-c-das**
**Change log description**
Upgrades the following libraries to higher versions to fix CVEs found.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/anju_das/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/anju_das/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Library | Current Version | Updated Version
-- | -- | --
jacksonVersion | 2.13.4 | 2.14.0
micrometerVersion | 1.2.0 | 1.3.7
com.fasterxml.woodstox:woodstox-core |   | 6.4.0

</body>

</html>

**Purpose of the change**
Fixes https://github.com/pravega/pravega/issues/6975

**What the code does**
Bumps up specified libraries to higher versions.

**How to verify it**
All tests must pass.

Following tests and compatibility have been done.

System test 1: 4871 successful.
System test 2 : 2337 successful.
SchemaRegistry : 53 successful.
PravegaSamples : 124 successful.
Pravega-flink-connector: 342 successful
Pravega-spark-connector : 75 successful.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -94,3 +94,5 @@ enableMkdocs=false
 # set simpleSnapshot during build to prevent getProjectVersion() from making
 # a complex -SNAPSHOT version string with commit count & commit hash
 #simpleSnapshot=true
+woodstox=6.4.0
```
Comment (kotlasaicharanreddy) [inline gradle.properties:97]: Please change this to `woodstoxVersion` and add it just after `jdomVersion`

Comment (anju-c-das) [reply gradle.properties]: Updated as suggested

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Thanks for the PR. For approving PRs with dependency changes, we normally do a safety check building other components of Pravega with the changes of this PR (connectors, samples, Schema Registry). This goes in addition to system tests. Please, let us know the results of this testing.

**Comment by anju-c-das** [comment]
> Thanks for the PR. For approving PRs with dependency changes, we normally do a safety check building other components of Pravega with the changes of this PR (connectors, samples, Schema Registry). This goes in addition to system tests. Please, let us know the results of this testing.

We have run the above-mentioned checks building other components and with System tests, the build numbers of which we have mentioned in the log description.
Please let me know if I have misunderstood above statement.

**Review by shshashwat — APPROVED** [review]
Based on the description, the changes looks good to be merged.

File: build.gradle [hunk 2]
```diff
@@ -139,6 +139,7 @@ allprojects {
 	    force "org.jetbrains.kotlin:kotlin-stdlib-common:" + kotlinVersion
 	    force "org.jetbrains.kotlin:kotlin-stdlib:" + kotlinVersion
 	    force "org.yaml:snakeyaml:" + snakeYamlVersion
+            force "com.fasterxml.woodstox:woodstox-core:" + woodstoxVersion
```
Comment (RaulGracia) [inline build.gradle:142]: Why do we need this new dependency? Just to force it to a version that does not contain CVE? What dependency is transitively importing this one?

**Review by RaulGracia — CHANGES_REQUESTED** [review]
It would be great to have a bit more clarification about the new dependency `woodstoxVersion`. ¡

File: build.gradle (hunk 2 above)
Comment (kotlasaicharanreddy) [reply build.gradle:142]: Azure is bringing this woodstox as a transitive dependency and it is present in pravega/lib folder as well. Yes just to force it to a version that doesn't contain a CVE.

**Review by kotlasaicharanreddy — APPROVED** [review]
LGTM

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**